src/invert/laplace/impls/serial_band/serial_band.cxx

```diff
@@ -106,21 +106,19 @@ const FieldPerp LaplaceSerialBand::solve(const FieldPerp &b, const FieldPerp &x0
     xend = mesh->LocalNx-2;
   }
 
-  for(int iz=0;iz<=ncz/2;iz++) {
+  for(int iz=0;iz<=maxmode;iz++) {
```

Comment:
That's a good point, it's not (not in `serial_tri` either) -- presumably something like

~~~~~
  for(int kz=maxmode+1; kz < ncz/2 + 1; kz++){
    for (int ix=0; ix<=ncx; ix++){
      xk[ix][kz] = 0.0;
    }
  }
~~~~~

after this loop would be an appropriate fix? 

Alternatively, we could do the initialisation at the point where we allocate the storage (i.e. in the constructor) such that this is only done once.